src/treebuilders/AdditionCalculator.h

```diff
@@ -30,26 +30,48 @@
 
 namespace mrcpp {
 
-template <int D> class AdditionCalculator final : public TreeCalculator<D> {
+template <int D, typename T> class AdditionCalculator final : public TreeCalculator<D, T> {
 public:
-    AdditionCalculator(const FunctionTreeVector<D> &inp)
-            : sum_vec(inp) {}
+    AdditionCalculator(const FunctionTreeVector<D, T> &inp, bool conjugate = false)
+        : sum_vec(inp),
+          conj(conjugate) {}
 
 private:
-    FunctionTreeVector<D> sum_vec;
+    FunctionTreeVector<D, T> sum_vec;
+    bool conj;
 
-    void calcNode(MWNode<D> &node_o) override {
+    void calcNode(MWNode<D, double> &node_o) {
```

Comment:
I believe we can keep the `override` marker here. We don't need to have two implementations for the different scalar types: one can use `if constexpr` to do compile-time, type-level checks nowadays. Of course it's going to be a bit of an ugly function...